src/openads/domain/service/DomainEventBus.js

```diff
@@ -0,0 +1,45 @@
+class DomainEventBus {
+  constructor () {
+    this._observers = new Map()
+  }
+
+  register ({eventName, observer}) {
+    if (!eventName) {
+      throw new Error('Event Name is required')
+    }
+    if (typeof observer !== 'function') {
+      throw new Error('Observer must be a function')
+    }
+    if (!this._observers.has(eventName)) {
+      this._observers.set(eventName, [observer])
+    } else {
+      this._observers.get(eventName).push(observer)
+    }
+  }
+
+  raise ({domainEvent}) {
+    this._observers
+      .get(domainEvent.eventName)
+      .forEach(observer => {
+        try {
+          observer({
+            payload: domainEvent.payload,
+            dispatcher: (data) => this.raise({domainEvent: data})
+          })
+        } catch (err) {
+          console.log('Error processing the observer: ', err)
```

Comment:
Yes, we were discussing this with Víctor, it should be shown an Error message using our Logger, the problem is that this class acts as a Singleton. An approach to solve this could be to instantiate it inside the Container class as an eager and pass the Logger using a constructor. What do you think?

Reply:
No, the idea of the DomainEventBus is to not have a logger in domain and use it to raise events throught all domain so if you want to log something that happens in domain, raise an event a log it from a observer of infrastructure.
Also you cannot inject an infrastructure in a Domain class